### PR TITLE
Part 3

### PR DESCRIPTION
**Pull request by jessibcoden**

### CODE REVIEW

**Comment by zoeames** [comment]
Authentication is good

**Comment by zoeames** [comment]
Your new page needs some more inputs - remember your json file has a contact that looks like this:

```
{
                  "first_name": "Kristin",
                  "last_name": "McKinney",
                  "title": "Recruiter/Partner Development",
                  "company": "NSS, Built",
      		"category": "Colleague",
      		"cell_phone": "(503)-799-8184",
      		"email": "kristin.mckinney@nashvillesoftwareschool.com",
                  "location": "East Nasty, TN",
                  "github_link": "none",
                  "linkedin_link": "https://www.linkedin.com/in/kristinmckinney/",
                  "image": "https://media-exp2.licdn.com/media/p/6/005/090/321/3bf926e.jpg",
                  "favorite": false,
                  "uid": "VNWDp1OvlVW4thyzkvuBGvkXIgi1"
		},
```

If i fill out that form you have i will only see:

```
{
                  "first_name": "Kristin",
                  "last_name": "McKinney",
                  "favorite": false,
                  "uid": "VNWDp1OvlVW4thyzkvuBGvkXIgi1"
		},
```

you need to add input fields and validation to create an ENTIRE contact

**Comment by zoeames** [comment]
Read and Delete are good just fix the new stuff above

**Comment by jessibcoden** [comment]
I did that to make testing easier.

**Comment by jessibcoden** [comment]
They're there, just commented out for now.

File: javascripts/controllers/NewCtrl.js [hunk 1]
```diff
@@ -1,5 +1,37 @@
...
+            "linkedin_link": contact.linkedin_link,
+            "image": contact.image,
+            "favorite": false,
+            "uid": userId
+        };
+    };
+
+    $scope.addNewContact = (contact) => {
+        let newContact = createContactObject(contact);
+        console.log("newContact", newContact);
```
Comment (zoeames) [inline javascripts/controllers/NewCtrl.js:28]: remove console.log

File: javascripts/controllers/NewCtrl.js [hunk 2]
```diff
@@ -1,5 +1,37 @@
...
+        ContactsService.postNewContact(newContact, contact.id).then((results) => {
```
Comment (zoeames) [inline javascripts/controllers/NewCtrl.js:29]: more stuff should be coming from your form or you will not have every attribute you need in createContactObject

File: javascripts/controllers/NewCtrl.js [hunk 3]
```diff
@@ -1,5 +1,37 @@
...
+            console.log("result", results);
```
Comment (zoeames) [inline javascripts/controllers/NewCtrl.js]: delete console log

File: javascripts/services/ContactsServices.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+                reject(err);
+            });
+        });
+    };
+
+    const deleteContact = (contactId) => {
+        return $http.delete(`${FIREBASE_CONFIG.databaseURL}/contacts/${contactId}.json`);
+    };
+
+
```
Comment (zoeames) [inline javascripts/services/ContactsServices.js]: delete extra line

File: partials/contacts/new.html [hunk 5]
```diff
@@ -3,99 +3,104 @@ <h1>New Contact</h1>
 </div>
 
 <div class="col-md-offset-1"> 
+    <form name="newContactForm" novalidate>
```
Comment (zoeames) [inline partials/contacts/new.html:6]: many more inputs needed on this page

File: partials/contacts/new.html [hunk 6]
```diff
@@ -3,99 +3,104 @@ <h1>New Contact</h1>
...
-            <fieldset class="form-group text-center">
-                <div class="col-md-4 inputGroupContainer">
-                    <div class="input-group">
-                        <span class="input-group-addon"><i class="fa fa-envelope-o"></i></span>
-                        <input name="email" placeholder="E-Mail Address" class="form-control"  type="text">
-                    </div>
-                    <div class="input-group">
-                        <span class="input-group-addon"><i class="fa fa-mobile"></i></span>
-                        <input name="phone" placeholder="(845)555-1212" class="form-control" type="text">
+        <!-- <fieldset class="container">
```
Comment (zoeames) [inline partials/contacts/new.html]: why is this code commented out?  comment it in or delete it

Comment (jessibcoden) [reply partials/contacts/new.html]: Commented them out for testing purposes so that you wouldn't have to enter all of those fields just to test.

**Comment by morecallan** [comment]
👍

**Pull request merged**